hazelcast/include/hazelcast/client/serialization/serialization.h

```diff
@@ -795,19 +817,22 @@ class HAZELCAST_API object_data_input
     typename std::enable_if<
       !(std::is_base_of<identified_data_serializer, hz_serializer<T>>::value ||
         std::is_base_of<portable_serializer, hz_serializer<T>>::value ||
+        std::is_base_of<compact_serializer, hz_serializer<T>>::value ||
         std::is_base_of<builtin_serializer, hz_serializer<T>>::value ||
         std::is_base_of<custom_serializer, hz_serializer<T>>::value),
       boost::optional<T>>::type inline read_object(int32_t type_id);
 
 private:
     pimpl::PortableSerializer& portable_serializer_;
+    pimpl::compact_stream_serializer& compact_serializer_;
     pimpl::DataSerializer& data_serializer_;
     std::shared_ptr<serialization::global_serializer> global_serializer_;
 };
 
 class HAZELCAST_API object_data_output : public pimpl::data_output
 {
     friend pimpl::DefaultPortableWriter;
+    friend pimpl::default_compact_writer;
```

Comment:
These two names seems a bit inconsistent, maybe we can change one of them? Or if you want ignore this comment